lib/auth/grpcserver.go

```diff
@@ -2445,6 +2446,29 @@ func (g *GRPCServer) GetNodes(ctx context.Context, req *types.ResourcesInNamespa
 	}, nil
 }
 
+// GetNodesStream returns a stream of all nodes in the given namespace.
+func (g *GRPCServer) GetNodesStream(req *types.ResourcesInNamespaceRequest, stream proto.AuthService_GetNodesStreamServer) error {
+	auth, err := g.authenticate(stream.Context())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	nodes, err := auth.ServerWithRoles.GetNodes(stream.Context(), req.Namespace)
```

Comment:
Do we care that the entire list of nodes is still being loaded into memory here, even though it is being streamed out piecemeal below?

Reply:
For this bug fix we don't care, but that would definitely be a good optimization to follow up with. Would you just query the database with pagination to achieve that?